aws/rust-runtime/aws-types/src/region.rs

```diff
@@ -70,19 +91,34 @@ impl ChainProvider {
 }
 
 impl ProvideRegion for Option<Region> {
-    fn region(&self) -> Option<Region> {
-        self.clone()
+    fn region(&self) -> RegionFuture {
+        RegionFuture::ready(self.clone())
     }
 }
 
 impl ProvideRegion for ChainProvider {
-    fn region(&self) -> Option<Region> {
-        for provider in &self.providers {
-            if let Some(region) = provider.region() {
-                return Some(region);
-            }
-        }
-        None
+    fn region(&self) -> RegionFuture {
+        RegionFuture::new(self.region())
+    }
+}
+
+pub struct RegionFuture<'a>(Pin<Box<dyn Future<Output = Option<Region>> + Send + 'a>>);
+
+impl<'a> RegionFuture<'a> {
+    pub fn new(f: impl Future<Output = Option<Region>> + Send + 'a) -> Self {
+        RegionFuture(Box::pin(f))
+    }
+
+    pub fn ready(region: Option<Region>) -> Self {
+        Self::new(std::future::ready(region))
```

Comment:
Consider adding a private enum to store a region directly vs. a future region and have `poll` match off this, which would allow you to avoid the boxed future in the case where a region is already set.